File: JokenPoBDR/src/entity/Player.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package entity;

import java.util.Scanner;

/**
 *A Classe Player tem como atributo nome e 
 *como métodos pode escolher armas e iniciar a batalha (onde pode usar pedra, papel ou tesoura)
 */
public class Player {
    
    private String nome;
    
    Pedra pedra;
    Papel papel;
    Tesoura tesoura;
   

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }
    
    
    
    public int escolheArma() {
        Scanner input = new Scanner(System.in);
        int armaUser;
        do {
            System.out.println("Escolha sua arma " + this.nome.toUpperCase() + "!\n" 
                    + "1 - Pedra\n"
                    + "2 - Papel\n"
                    + "3 - Tesoura");
            System.out.print("Sua arma será → ");
            armaUser = input.nextInt();             

            if (armaUser > 3 || armaUser < 1) {
                System.out.println("Essa arma não existe neste mundo... Tente Novamente!");
            }

        } while (armaUser > 3 || armaUser < 1);
        
        return armaUser;
    }
    
    public void vamosBatalhar(int arma1, int arma2) {
        System.out.println("Agora vamos ao combate!!");         
        switch (arma1) {
            case 1:
                pedra.usaPedra(arma2, nome);
                break;

            case 2:
                papel.usaPapel(arma2, nome);
                break;

            case 3:
                tesoura.usaTesoura(arma2, nome);
                break;
        }
    }
    
    
    
    
    
}

File: JokenPo_Python/Player.py
from JokenPo_Python.Papel import Papel
from JokenPo_Python.Pedra import Pedra
from JokenPo_Python.Tesoura import Tesoura


class Player:
    def __init__(self, nome):
        self.nome = nome

    def escolheArma(self):
        armaUser1 = input("Escolha sua arma:\n 1 - Pedra\n 2 - Papel \n 3 - Tesoura \n →→ ")
        while (int(armaUser1) < 1 or int(armaUser1) > 3):
            print("Não existe esta arma, tente novamente!")
            armaUser1 = input("Escolha sua arma:\n 1 - Pedra\n 2 - Papel \n 3 - Tesoura \n →→ ")
        return armaUser1

    def vamosBatalhar(self, arma1, arma2):
        if int(arma1) == 1:
            Pedra.usaPedra(int(arma2), self.nome.upper())
        elif int(arma1) == 2:
            Papel.usaPapel(int(arma2), self.nome.upper())
        elif int(arma1) == 3:
            Tesoura.usaTesoura(int(arma2), self.nome.upper())
